ql/src/java/org/apache/hadoop/hive/ql/optimizer/calcite/rules/HiveJoinConstraintsRule.java

```diff
@@ -334,11 +335,13 @@ private void rewrite(final Mode mode, final RelNode inputToKeep, final RelNode i
       }
     } else { // Mode.TRANSFORM
       // Trigger transformation
+      List<RexNode> exps = new ArrayList<>();
+      project.accept(new ChildExpsRexShuttle(exps));
       call.transformTo(call.builder()
           .push(leftInput).push(rightInput)
           .join(JoinRelType.INNER, join.getCondition())
           .convert(call.rel(1).getRowType(), false) // Preserve nullability
-          .project(project.getChildExps())
```

Comment:
Why not `project.getProjects()` as in other places?

Reply:
The project here is of type `RelNode` which doesn't have a `getProjects` method. In other places, generally the variable is of `Project` type.